libs/mailroom/ngx/src/lib/modules/list/list.component.ts

```diff
@@ -28,14 +35,24 @@ export class ListComponent implements OnInit {
   }
 
   public deleteEmail(email: MailroomEmail) {
-    this.emailService.deleteEmail(email.id).subscribe((removed) => {
-      if (removed) {
-        this.ns.preset('deleted_email_success');
-
-        this._$refresh.next(true);
-      } else {
-        this.ns.preset('deleted_email_failure');
-      }
-    });
+    this.modal
+      .open<{ deleteEmail: boolean }>(DeleteEmailModalComponent, {
+        data: {
+          message: email.id
+        }
+      })
+      .subscribe((proceedWithDelete) => {
+        if (proceedWithDelete) {
+          this.emailService.deleteEmail(email.id).subscribe((removed) => {
+            if (removed) {
+              this.ns.preset('deleted_email_success');
+
+              this._$refresh.next(true);
+            } else {
+              this.ns.preset('deleted_email_failure');
+            }
```

Comment:
Fail case is never being reached if the http request returns a server error code.

Use observable subscribe and error callbacks:

https://blog.angular-university.io/rxjs-error-handling/


Should also probably explictly state the return type oft the `emailService.deleteEmail`. Right now it's an implicit `Observable<Object>` which is wrong since I now know it's meant to be `Observable<boolean>`.

Reply:
Fixed.